src/openforms/emails/digest.py

```diff
@@ -389,16 +417,30 @@ def collect_invalid_logic_rules() -> list[InvalidLogicRule]:
 
         form_logics_vars = []
         for logic in form_logics:
-            form_logics_vars += introspect_json_logic(
-                logic.json_logic_trigger
-            ).get_input_keys()
+            logic_introspection_result = introspect_json_logic_wrapper(
+                logic.json_logic_trigger, form.admin_name
+            )
+
+            if logic_introspection_result is None:
+                _report(exception=True)
+
+                # no need for checking actions since the logic rule is already broken
+                continue
+            else:
+                form_logics_vars += logic_introspection_result
```

Comment:
you can drop the `else` and remove a level of indentation - the `continue` above already skips the rest of the loop execution.